src/main/java/org/alfasoftware/soapstone/SoapstoneOpenApiReader.java

```diff
@@ -413,6 +423,40 @@ private RequestBody parametersToRequestBody(String operationId, Collection<Param
     JavaType javaType = soapstoneConfiguration.getObjectMapper().constructType(type);
     LOG.debug("          " + javaType.toString());
 
+    XmlJavaTypeAdapter adapterAnnotation = currentResourceClass.getPackage().getAnnotation(XmlJavaTypeAdapter.class);
```

Comment:
I think we should probably extract the package annotation lookup to its own method and add some comments to explain why we need to do this here.

Reply:
having looked at this, it seemed to make most sense to me to extract the converter lookup into a private method but this may not be what you had in mind. I have pushed the change, if my refactor and the comments are nonsensical then a quick chat would be good as that means I've misunderstood something.

Reply:
@willhowes - not quite, no. It's a bit odd that we get the adapter annotation here, and then have a private method just for handling the subsequently retrieved array-holding annotation. Well, perhaps not odd so much as a bit arbitrary. Let's just move _all_ the logic for getting hold of the converter into the private method.